ink-wrapper/src/codegen.rs

```diff
@@ -275,30 +346,40 @@ fn define_reader(
     message: &MessageSpec<PortableForm>,
     visibility: &str,
     metadata: &InkProject,
-) -> rust::Tokens {
-    let data = &new_name("data", message.args());
+) -> proc_macro2::TokenStream {
+    let data_ident = &new_name("data", message.args());
+    let docs = docs(message.docs());
+    let reader_head = define_reader_head(message, visibility, metadata);
+    let args = gather_args(message.selector().to_bytes(), message.args());
 
     quote! {
-        $(docs(message.docs()))
+        #docs
         #[allow(dead_code, clippy::too_many_arguments)]
-        $(define_reader_head(message, visibility, metadata))
+        #reader_head
         {
-            let $(data) = $(gather_args(message.selector().to_bytes(), message.args()));
-            ink_wrapper_types::ReadCall::new(self.account_id, $(data))
+            let #data_ident = #args;
+            ink_wrapper_types::ReadCall::new(self.account_id, #data_ident)
         }
-
-        $[ '\n' ]
     }
 }
 
 fn define_reader_head(
     message: &MessageSpec<PortableForm>,
     visibility: &str,
     metadata: &InkProject,
-) -> rust::Tokens {
+) -> proc_macro2::TokenStream {
+    let method_name = format_ident!("{}", message.method_name());
+    let args = message_args(message.args(), metadata);
+    let read_call_type = type_ref(message.return_type().opt_type().unwrap().ty().id, metadata);
+    let visibility = if visibility.is_empty() {
+        quote! {}
+    } else {
+        let v = format_ident!("{}", visibility);
+        quote! { #v }
+    };
```

Comment:
Suggestion: extract method.
```suggestion
    let visibility = quote_visiblity(visibility);
```

Reply:
Done in [review](https://github.com/Cardinal-Cryptography/ink-wrapper/pull/62/commits/adbdce3650a745ea28587d26251e881765f3d9f0)